core/src/array/array_schema.cc

```diff
@@ -861,14 +847,7 @@ int64_t ArraySchema::tile_num() const {
     return tile_num<uint32_t>();
   else if (types_[attribute_num_] == TILEDB_UINT64)
     return tile_num<uint64_t>();
-
   assert(0);
-  std::string errmsg =
-      "Unsupported dimensions type for retrieving the "
-      "number of tiles";
-  PRINT_ERROR(errmsg);
-  tiledb_as_errmsg = TILEDB_AS_ERRMSG + errmsg;
-  return TILEDB_AS_ERR;
```

Comment:
this helps suppress warnings.  These should be enums so that the compiler can do exhaustive switch case testing.